tools/scripts/enterprise_governance_gate.py

```
@@ -158,10 +219,147 @@ def _fail(msg: str) -> None:
+    
+    Parameters:
+        cmd (list[str]): Command to run; if `use_shell` is True the list is joined into a single shell string.
+        env (dict[str, str]): Environment variables to use for the subprocess.
+        use_shell (bool): If True run the command through the shell (joins `cmd` into one string).
+        timeout (int): Maximum time in seconds to allow the subprocess to run before raising a timeout.
+    
+    Returns:
+        subprocess.CompletedProcess[str]: CompletedProcess containing `args`, `returncode`, `stdout`, and `stderr`.
+    """
+    if use_shell:
+        return subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)